Merge XML Format support

Lokesh-K-Haralakatta opened the pull request:
The code changes include -
- XML Codec using dicttoxml and xmltodict modules
- Utility code to contain common methods for the library
- Required changes to device, application and gateway code
- Added required unit tests

The tests are run on jenkins to make sure existing functionality is not regressed

durera commented:
Rather than create new ibmiotf.util, can you move the `getContentType()` method into existing `ibmiotf.HttpAbstractClient` class, as it's only the client httpClient's which need that method I'll pull everything in to master.

Lokesh-K-Haralakatta commented:
As of now, we have only 1 util method, but going forward, the library may need different util methods, we can group them under one ibmiotf.util file for easy reference in the code.

Lokesh-K-Haralakatta commented:
I have moved the getContentType method to ibmiotf.HttpAbstractClient class and removed util.py file. Updated the source files and tests accordingly, executed tests to make sure the changes work fine. Can you please have a look and merge?
